google-cloud-clients/google-cloud-core/src/main/java/com/google/cloud/Binding.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.core.InternalApi;
+import com.google.common.base.MoreObjects;
+import java.util.*;
+
+public class Binding {
+  private Role role;
+  private Set<Identity> identities;
+  private Condition condition;
+
+  public static class Builder {
+    private Role role;
+    private Set<Identity> identities;
+    private Condition condition;
+
+    @InternalApi("This class should only be extended within google-cloud-java")
+    protected Builder() {}
+
+    @InternalApi("This class should only be extended within google-cloud-java")
+    protected Builder(Binding binding) {
+      setRole(binding.role);
+      setIdentities(binding.identities);
+      setCondition(binding.condition);
+    }
+
+    public final Binding.Builder setRole(Role role) {
+      this.role = role;
+      return this;
+    }
+
+    public final Binding.Builder setIdentities(Set<Identity> identities) {
+      this.identities = identities;
+      return this;
+    }
+
+    public final Binding.Builder setCondition(Condition condition) {
+      this.condition = condition;
+      return this;
+    }
+
+    /**
+     * Adds one or more identities to the binding
+     *
+     * @throws NullPointerException if any of the identities is null.
+     */
+    public final Builder addIdentity(Identity first, Identity... others) {
+      String nullIdentityMessage = "Null identities are not permitted.";
+      checkNotNull(first, nullIdentityMessage);
+      checkNotNull(others, nullIdentityMessage);
+      for (Identity identity : others) {
+        checkNotNull(identity, nullIdentityMessage);
+      }
+      Set<Identity> toAdd = new LinkedHashSet<>();
+      toAdd.add(first);
+      toAdd.addAll(Arrays.asList(others));
+      if (identities == null) {
```

Comment:
Might be better to move this to the constructor, would also allow you to skip the null check below